test/integration/agent/networkpolicy_metrics_test.go

```diff
@@ -0,0 +1,206 @@
+// Copyright 2020 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package agent
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/apimachinery/pkg/watch"
+	k8stesting "k8s.io/client-go/testing"
+	"k8s.io/component-base/metrics/legacyregistry"
+	"k8s.io/component-base/metrics/testutil"
+
+	"github.com/vmware-tanzu/antrea/pkg/agent/controller/networkpolicy"
+	"github.com/vmware-tanzu/antrea/pkg/agent/metrics"
+	"github.com/vmware-tanzu/antrea/pkg/apis/networking/v1beta1"
+	"github.com/vmware-tanzu/antrea/pkg/client/clientset/versioned"
+	"github.com/vmware-tanzu/antrea/pkg/client/clientset/versioned/fake"
+)
+
+const networkPolicyTimeout = time.Millisecond * 100
+const networkPolicyInterval = time.Millisecond * 5
+
+const testNamespace = "ns1"
+
```

Comment:
we usually group `const` declarations

Reply:
> Can you remind me why this is an integration test and not a plain unit test? Is there some system dependency I am missing?

@srikartati  Please correct me if I'm wrong.
I think it's because we want to validate metrics with a running agent networkpolicy controller.
As for networkpolicy metrics in this PR, I'd like to check if they are accurate after networkpolicies are added or deleted so I created this as an integration test.

Reply:
But if we don't consider testing these metrics in networkpolicy_controller_test.go as an integration test, I think I have to use a unit test instead.
Please give me your opinion. I also tested metrics in networkpolicy_controller_test.go and I didn't find any trouble so either will do.

Reply:
> > Can you remind me why this is an integration test and not a plain unit test? Is there some system dependency I am missing?
> 
> @srikartati Please correct me if I'm wrong.
> I think it's because we want to validate metrics with a running agent networkpolicy controller.
> As for networkpolicy metrics in this PR, I'd like to check if they are accurate after networkpolicies are added or deleted so I created this as an integration test.

Thanks for bringing this up @antoninbas 
@yktsubo  My previous tests for other agent metrics were piggybacked on existing integration tests, where OVS is needed. In the case of network policy metrics, as there is no such requirement this test can be moved to networkpolicy_controller_test.go as a unit test.

Reply:
Thank you so much for your feedback. @srikartati @antoninbas 
I'll put this test into unit tests.